File: src/ticketflow/ticket.py
from typing import List


class Ticket:
    """Represents a support ticket"""
    
    def __init__(self, id: str, title: str, description: str, category: str, priority: str, status: str, created_at: str, updated_at: str, description_vector: List[float]):
        self.id = id
        self.title = title
        self.description = description
        self.category = category
        self.priority = priority
        self.status = status
        self.created_at = created_at
        self.updated_at = updated_at
        self.description_vector = description_vector

File: frontend/ticketflow-client/src/types/ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  category: string;
  priority: "low" | "medium" | "high" | "urgent";
  status: "new" | "processing" | "resolved" | "escalated" | "closed";
  user_id: string;
  user_email: string;
  user_type: string;
  resolution: string;
  resolved_by: string;
  resolution_type: "automated" | "human" | "escalated";
  agent_confidence: number;
  processing_duration_ms: number;
  similar_cases_found: number;
  kb_articles_used: number;
  workflow_steps: WorkflowStep[];
  ticket_metadata: Record<string, any>;
  created_at: string;
  updated_at: string;
  resolved_at?: string;
}

export interface TicketCreateRequest {
  title: string;
  description: string;
  category: string;
  priority?: "low" | "medium" | "high" | "urgent";
  user_id?: string;
  user_email?: string;
  user_type?: string;
  ticket_metadata?: Record<string, any>;
}

export interface WorkflowStep {
  step_name: string;
  step_type: string;
  started_at: string;
  completed_at?: string;
  duration_ms: number;
  status: string;
  input_data: Record<string, any>;
  output_data: Record<string, any>;
  error_message: string;
}